### PR TITLE
[ml] Improve the documentation of ml common classes; add kwargs to predictor

### PR DESCRIPTION
**Pull request by ericl**
This PR improves the doc quality of these common classes.

It also adds kwargs to predictor in order to support customization of predictions (e.g., specification of label column). This one I think needs a little discussion.

### CODE REVIEW

**Comment by richardliaw** [comment]
> It also adds kwargs to predictor in order to support customization of predictions (e.g., specification of label column). This one I think needs a little discussion.

Is there a viable alternative?

**Comment by krfricke** [comment]
For the predictor kwargs it would be good to see how e.g. Serve calls it in order to see if there are sensible alternatives. Otherwise looks good to me.

File: python/ray/ml/preprocessor.py [hunk 1]
```diff
@@ -9,15 +9,25 @@
...
 
 class PreprocessorNotFittedException(RuntimeError):
+    """Error raised when the preprocessor needs to be fitted first."""
+
     pass
 
 
 class Preprocessor(abc.ABC):
-    """Preprocessor interface for transforming Datasets."""
+    """Implements a ML preprocessing operation.
```
Comment (matthewdeng) [inline python/ray/ml/preprocessor.py:24]: nit:
```suggestion
    """Implements an ML preprocessing operation.
```

**Review by matthewdeng — APPROVED** [review]
Preprocessor docs LGTM, thanks for doing this!

**Pull request merged**